src/EventCalendar.js

```diff
@@ -5,58 +5,97 @@
 'use strict';
 
 require('jquery');
+import $ from 'jquery';
 var moment = require('moment');
+require('moment-recur');
 window._ = require('underscore');
 require('clndr');
 
 class EventCalendar {
 
     /**
-     * EventCalendar
      * @constructor
-     * @param {jQuery} $html - jQuery wrapper of the html node
+     * @param {jQuery} $html jQuery wrapper of the html node
+     * @param {jQuery} clndr The clndr instance that is in use
      */
     constructor ($html) {
         this.$html = $html;
+        this.$patternField;
+        this.$ariaLiveRegion;
+        this.dateFormat = 'YYYY-MM-DD';
+        this.dateFormatLong = 'D MMMM, YYYY';
+        this.clndr;
     }
 
     /**
-     * Initialise
+     * Initalises the event calendar providing the correct markup is present within the DOM (as per README.md).
+     * 
+     * @param {string} startDate Specifies the month and year which will be initially focused by the calendar, dates 
+     * before this will not be interactive. Will default to the user's current date if not supplied or is null 
+     * (format: YYYY-MM-DD)
+     * @param {string} endDate Specififies the end of dates made available to the user. Defaults to 
+     * startDate + 15 years if not supplied. (format: YYYY-MM-DD)
+     * @param {array} events An array of simple dates that should be highlighted in the calendar as selected
+     *
+     * let exampleEvents = [
+     *    { date: 'YYYY-MM-DD' },
+     *    { date: 'YYYY-MM-DD' },
+     *    { date: 'YYYY-MM-DD' }
+     * ];
      */
-    init (events) {
+    init (startDate, endDate, events) {
         if (typeof this.$html === 'undefined' || !this.$html.length) {
             throw new Error('$html must be passed to EventCalendar');
+        }   
+
+        let $container = this.$html.find('.js-event-calendar');
+
+        if (typeof $container === 'undefined' || !$container.length) {
+            throw new Error('EventCalendar requires a .js-event-calendar element present in the DOM');
         }
 
         let _self = this,
```

Comment:
Admittedly this is a really old habit I picked up somewhere, but what would you do around L:171 where `this` changes scope? I've always stored a ref like this so in my own head I know where I am.

Reply:
yeah it's a really valid point, it's habitual for a lot of people because in all honesty in a lot of situations it makes scope easier to reason about. I think with the introduction of arrow functions (ES6) `() => {}` (which will use the "lexical" `this` – the context of the block containing the function) it makes life a little easier. 

For the line `171` example we could use a named function and bind the context manually – this is why many people prefer to use `var that = this;`, because it's quicker and easier in a lot of cases.

```
class Paul {
    sayHello () {
        $foo.on('click', handleClick.bind(this))        
    }

    handleClick () {
        console.log(this) // Paul instance
    }
}
```

or with arrow functions...

```
class Paul {
    sayHello () {
        $foo.on('click', () => {
            console.log(this) // Paul instance
        })        
    }
}
```